hive-metastore/src/test/java/org/apache/iceberg/hive/HiveMetastoreExtension.java

```diff
@@ -19,28 +19,22 @@
 package org.apache.iceberg.hive;
 
 import java.util.Map;
-import java.util.concurrent.TimeUnit;
 import org.apache.hadoop.hive.conf.HiveConf;
 import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
 import org.apache.hadoop.hive.metastore.api.Database;
-import org.apache.iceberg.CatalogProperties;
-import org.apache.iceberg.CatalogUtil;
-import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
 import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 import org.junit.jupiter.api.extension.AfterEachCallback;
 import org.junit.jupiter.api.extension.BeforeEachCallback;
 import org.junit.jupiter.api.extension.ExtensionContext;
 
-public final class HiveMetastoreExtension implements AfterEachCallback, BeforeEachCallback {
-
-  static HiveCatalog catalog;
-  static HiveMetaStoreClient metastoreClient;
-  static TestHiveMetastore metastore;
-  static HiveConf hiveConf;
+public class HiveMetastoreExtension implements BeforeEachCallback, AfterEachCallback {
+  private HiveMetaStoreClient metastoreClient;
+  private TestHiveMetastore metastore;
   private final Map<String, String> hiveConfOverride;
-  static final String DB_NAME = "hivedb";
+  private final String databaseName;
 
-  public HiveMetastoreExtension(Map<String, String> hiveConfOverride) {
+  public HiveMetastoreExtension(String databaseName, Map<String, String> hiveConfOverride) {
```

Comment:
Thanks for looking into it. I had plan to do the same with followup patch . 
But I was more considering like multiple constructor one with only hive config, other with db and table name. The same can help to intialize the extended classed of [HiveTableBaseTest](https://github.com/apache/iceberg/blob/4433aa8c62002e2f3c589c7d3d8ef61dee8284b1/hive-metastore/src/test/java/org/apache/iceberg/hive/HiveTableBaseTest.java#L60)

Reply:
I think it makes sense now to do the setup only with `HiveMetastoreExtension` all the other catalog initialisation or default table creation can happen with usage basis.